### ISSUE TITLE
Buggy depreciation warining (client_factory will be replaced by gspread.http_client types)

### ISSUE THREAD

**Issue by ArturoGallettiLopez** [burnash/gspread#1348]
I am currently cleaning up my code for version 6.0 and ran into a somewhat strange scenario

When calling service_account_from_dict in a subfunction like so:

```python
#imports
import gspread
from gspread import service_account_from_dict
#function declare
def BugTest():
    gc = service_account_from_dict(creds1, scope)

#run
BugTest()
```

I get the following depreciation warning:
 DeprecationWarning: [Deprecated][in version 6.0.0]: client_factory will be replaced by gspread.http_client types  BugTest()

If I run the same code as a standalone I do not get any warnings and the rev notes don't seem to indicate any particular issue.

**Comment by alifeee** [comment]
strange indeed...

thanks for the issue :)

I do not know the subtleties of the `warnings` package, but my guess is that it has something to do with the `stacklevel` kwarg.

This is how we emit a warning:

https://github.com/burnash/gspread/blob/67759c7ffe88c0a4650d1eaf8bea20156c7d0e31/gspread/utils.py#L926-L930

The third argument to `warnings.warn(` is `stacklevel` (see [`warnings.warn` docs](https://docs.python.org/3/library/warnings.html#warnings.warn)). I'm not fully sure what this means, but it seems that it only warns if the stack level is four or more (or something).

In your case, the stack level increases when you wrap `service_account_from_dict` in a function, which allows the warning to show. Maybe we need to increase/decrease the `stacklevel` argument so the warning always shows. maybe @lavigne958 can provide more info, having implemented the warnings.

As an addendum, I have made a PR #1349 which disables this warning if you have not set the `client_factory` kwarg in `service_account_from_dict`. In your case, the warning is not very useful, since you do not use the `client_factory` kwarg. This should hide the warning for you when 5.12.1 releases.

For now, as well, if you want to hide warnings, you can set the `SILENCE_WARNINGS_ENV_KEY` environment variable to "1" to disable warnings.

**Comment by ArturoGallettiLopez** [comment]
I'm going to create the gc object one level to take care of it in my code. appreciate the prompt response.

**Comment by lavigne958** [comment]
Hi thank you for raising this issue.

Yes this is due to the stack leve.
The stack level is: the number of function calling the current code.
Stack levels start at 0 from the line of code in the warning package.

Meaning:
in the deprecation warnign I set a stack to 4, so when the warning is emited, it prints as well the code where the user is using some gspread code. 
As mentioned in the comment: staick 0 -> is warning package, 1 -> itl function in gspread, up to 3 (or 4 in 1 based notation) which the line of code from the user. so it prints the file/line where someone is calling a gpsread function which triggers the warning..... in most of the case ! sometimes it could be 1 stack above or 1 lower :-( 

That should do for now it's only informative, we'll rlease 6.0.0 soon (:christmas_tree: :question:  :upside_down_face: ) so it won't last.

**Comment by alifeee** [comment]
this is fixed by #1349 and will be included in the next release, 5.12.1
